pkg/api/v1/schema.go

```diff
@@ -35,8 +35,20 @@ type DomainSpec struct {
 }
 
 type Memory struct {
-	Value uint   `json:"value"`
-	Unit  string `json:"unit"`
+	// Allocated field defines the amount of memory a VM gets upon starting in kiloBytes
+	// The field type means the total maximum is 4 TiB of memory
+	Allocated uint `json:"allocated"`
+
+	// Required fields defines the amount of memory a VM needs to operate properly.
+	// This field can be set to the same value as `Allocated` to effectively
+	// disable any memory overcommit features. The value is provided in Bytes.
+	Required uint `json:"required"`
+
+	// Maximum defines the maximum amount of memory a VM can ever get in kiloBytes.
+	// This field depends on qemu capabilities, but can be only 4x as big as
+	// `Allocated` at the time I am writing this (8/2017)
+	// The field type means the total maximum is 4 TiB of memory
+	Maximum uint `json:"maximum"`
```

Comment:
On `allocated`, `required`, and `maximum` - Did yo uconsider to stick to the `limits` and `requests` terminology as kube is using it?

Reply:
They are missing the `required` equivalent. allocated can be mapped to requests, maximum to limits, but balloon controller (which we do not have atm) needs to know how much memory it is allowed to remove from the VM.

Reply:
the least interesting is the "maximum"...so how about mapping requests to min memory guaranteed and limits to current memory, and keep maximum (by default 4x current) just in kubevirt's VM spec in maxMemory element?